examples/kcp/main.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	kcpclient "github.com/kcp-dev/apimachinery/pkg/client"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	api "sigs.k8s.io/controller-runtime/examples/crd/pkg"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var (
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+type reconciler struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx).WithValues("chaospod", req.ObjectKey)
+	log.V(1).Info("reconciling chaos pod")
+
+	fmt.Println("***************************************")
+	fmt.Println(req.ObjectKey.Cluster)
+	fmt.Println(ctx.Value("clusterName"))
+	fmt.Println("***************************************")
+	// log.Info(fmt.Sprintf("%+v\n\n%+v\n", ctx, req))
+
+	var chaospod api.ChaosPod
+	if err := r.Get(ctx, req.ObjectKey, &chaospod); err != nil {
+		log.Error(err, "unable to get chaosctl")
+		return ctrl.Result{}, err
+	}
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-cm",
+			Namespace: "default",
+			// ClusterName: req.ClusterName,
+		},
+		Data: map[string]string{
+			"test-key": "test-value",
+		},
+	}
+	if err := r.Create(ctx, cm); err != nil {
+		log.Info("CM creation failed", err)
+		// log.Error(err, "this is fine")
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func main() {
+
+	ctrl.SetLogger(zap.New())
+
+	cfg := ctrl.GetConfigOrDie()
+	httpClient, err := rest.HTTPClientFor(cfg)
```

Comment:
should probably make a helper for this, like kcpclient.NewClientForConfig(cfg) that gives you an http.Client with the transport set.